homeassistant/components/aws/image_processing.py

```diff
@@ -0,0 +1,483 @@
+"""AWS platform for image_processing component."""
+import io
+import logging
+import re
+
+from PIL import Image, ImageDraw, UnidentifiedImageError
+import aiobotocore
+
+from homeassistant.components.image_processing import (
+    ATTR_AGE,
+    ATTR_CONFIDENCE,
+    ATTR_GENDER,
+    ATTR_GLASSES,
+    ATTR_MOTION,
+    ATTR_NAME,
+    CONF_CONFIDENCE,
+    CONF_ENTITY_ID,
+    CONF_NAME,
+    CONF_SOURCE,
+    ImageProcessingEntity,
+    ImageProcessingFaceEntity,
+)
+from homeassistant.const import ATTR_ENTITY_ID, CONF_PLATFORM
+from homeassistant.core import split_entity_id
+import homeassistant.util.dt as dt_util
+from homeassistant.util.pil import draw_box
+
+from .const import (
+    ATTR_LABELS,
+    ATTR_OBJECTS,
+    CONF_ACCESS_KEY_ID,
+    CONF_COLLECTION_ID,
+    CONF_CREDENTIAL_NAME,
+    CONF_DETECTION_ATTRIBUTES,
+    CONF_IDENTIFY_FACES,
+    CONF_PROFILE_NAME,
+    CONF_REGION,
+    CONF_SAVE_FILE_FOLDER,
+    CONF_SAVE_FILE_TIMESTAMP,
+    CONF_SECRET_ACCESS_KEY,
+    CONF_SERVICE,
+    DATA_SESSIONS,
+    EVENT_LABEL_DETECTED,
+    EVENT_OBJECT_DETECTED,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DATETIME_FORMAT = "%Y-%m-%d_%H:%M:%S"
+
+
+async def get_valid_filename(name):
+    """Parse input to ensure valid filename characters."""
+    return re.sub(r"(?u)[^-\w.]", "", str(name).strip().replace(" ", "_"))
+
+
+async def save_image(
+    self, image, entity_name, objects, directory, save_timestamp=False
+):
+    """Draw the bounding boxes and save the image."""
+    try:
+        img = Image.open(io.BytesIO(bytearray(image))).convert("RGB")
+    except UnidentifiedImageError:
+        _LOGGER.warning("Rekognition unable to process image, bad data")
+        return
+    draw = ImageDraw.Draw(img)
+    for obj in objects:
+        name = obj["name"]
+        confidence = obj["confidence"]
+        box = obj["bounding_box"]
+        centroid = obj["centroid"]
+        box_colour = (255, 255, 0)  # Yellow
```

Comment:
Yellow could be added to `homeassistant/util/pil.py`

Reply:
Adding this constant along with other basic colours to `utils/pil.py` - can easily revert if needed